lib/jazzy/doc_builder.rb

```diff
@@ -235,6 +238,45 @@ def self.document_markdown(source_module, doc_model, path_to_root)
       doc.render
     end
 
+    # Returns the appropriate color for the provided percentage,
+    # used for generating a badge on shields.io
+    # @param [Number] coverage The documentation percentage
+    def self.color_for_coverage(coverage)
+      if coverage < 10
+        return 'red'
+      elsif coverage < 30
+        return 'orange'
+      elsif coverage < 60
+        return 'yellow'
+      elsif coverage < 85
+        return 'yellowgreen'
+      elsif coverage < 90
+        return 'green'
+      else
+        return 'brightgreen'
+      end
+    end
+
+    # Downloads an SVG from shields.io displaying the documentation percentage
+    # @param [SourceModule] source_module The source module from SourceKitten
+    # @param [Config] options Build options
+    def self.download_badge(source_module, options)
+      if options.hide_documentation_coverage then
+        return
+      end
+      color = self.color_for_coverage(source_module.doc_coverage)
+      uri = URI.parse("https://img.shields.io")
+      url_path = "/badge/documentation-" +
+                 "#{source_module.doc_coverage}%25-#{color}.svg"
+      Net::HTTP.start(uri.host, uri.port, :use_ssl => true) do |http|
```

Comment:
I've said this before, but I think the build should fail if we expected to be able to generate the badge but that couldn't succeed. Inevitably, users will expect the badge to be there if the build succeeded.